ksp_plugin/pile_up.cpp

```diff
@@ -41,12 +43,38 @@ using physics::RigidMotion;
 using quantities::Angle;
 using quantities::AngularFrequency;
 using quantities::AngularMomentum;
+using quantities::Inverse;
 using quantities::Time;
+using quantities::si::Kilogram;
+using quantities::si::Metre;
 using quantities::si::Radian;
+using quantities::si::Second;
 using ::std::placeholders::_1;
 using ::std::placeholders::_2;
 using ::std::placeholders::_3;
 
+// NOTE(phl): The default values were tuned on "Fubini oscillator".
+constexpr double kp_default = 0.65;
+constexpr Inverse<Time> ki_default = 0.03 / Second;
+constexpr Time kd_default = 0.0025 * Second;
+
+// Configuration example:
+//   principia_flags
+//   {
+//     kp = 0.65
+//     ki = 0.03 / s
+//     kd = 0.0025 s
+//   }
+template<typename Q>
+Q GetPidFlagOr(std::string_view const name, Q const default) {
```

Comment:
Principia style is to keep acronyms in upper case in camel case identifiers, see `ICRS`, `ParseTT`, `ParseTAI`, etc. This should say PID.

I would want this to return an optional and drop the `Or` but obviously this does not work because we cannot deduce Q, unless we build a mechanism to declare flags. Open an issue to track that, we don’t need it yet but we likely will at some point, the flags are inevitably becoming useful.

Reply:
#2551